crates/ruma-common/tests/events/poll.rs

```diff
@@ -0,0 +1,468 @@
+#![cfg(feature = "unstable-msc3381")]
+
+use std::convert::TryInto;
+
+use assign::assign;
+use js_int::uint;
+use matches::assert_matches;
+use ruma_common::{
+    event_id,
+    events::{
+        message::MessageContent,
+        poll::{
+            end::{PollEndContent, PollEndEventContent},
+            response::{PollResponseContent, PollResponseEventContent},
+            start::{
+                PollAnswer, PollAnswers, PollAnswersError, PollKind, PollStartContent,
+                PollStartEventContent,
+            },
+            ReferenceRelation,
+        },
+        AnyMessageLikeEvent, MessageLikeEvent, MessageLikeUnsigned, OriginalMessageLikeEvent,
+    },
+    room_id, user_id, MilliSecondsSinceUnixEpoch,
+};
+use serde_json::{from_value as from_json_value, json, to_value as to_json_value};
+
+#[test]
+fn poll_answers_deserialization_valid() {
+    let json_data = json!([
+        { "id": "aaa", "m.text": "First answer" },
+        { "id": "bbb", "m.text": "Second answer" },
+    ]);
+
+    assert_matches!(
+        from_json_value::<PollAnswers>(json_data),
+        Ok(answers) if answers.answers().len() == 2
+    );
+}
+
+#[test]
+fn poll_answers_deserialization_truncate() {
+    let json_data = json!([
+        { "id": "aaa", "m.text": "1st answer" },
+        { "id": "bbb", "m.text": "2nd answer" },
+        { "id": "ccc", "m.text": "3rd answer" },
+        { "id": "ddd", "m.text": "4th answer" },
+        { "id": "eee", "m.text": "5th answer" },
+        { "id": "fff", "m.text": "6th answer" },
+        { "id": "ggg", "m.text": "7th answer" },
+        { "id": "hhh", "m.text": "8th answer" },
+        { "id": "iii", "m.text": "9th answer" },
+        { "id": "jjj", "m.text": "10th answer" },
+        { "id": "kkk", "m.text": "11th answer" },
+        { "id": "lll", "m.text": "12th answer" },
+        { "id": "mmm", "m.text": "13th answer" },
+        { "id": "nnn", "m.text": "14th answer" },
+        { "id": "ooo", "m.text": "15th answer" },
+        { "id": "ppp", "m.text": "16th answer" },
+        { "id": "qqq", "m.text": "17th answer" },
+        { "id": "rrr", "m.text": "18th answer" },
+        { "id": "sss", "m.text": "19th answer" },
+        { "id": "ttt", "m.text": "20th answer" },
+        { "id": "uuu", "m.text": "21th answer" },
+        { "id": "vvv", "m.text": "22th answer" },
+    ]);
+
+    assert_matches!(
+        from_json_value::<PollAnswers>(json_data),
+        Ok(answers) if answers.answers().len() == 20
+    );
+}
+
+#[test]
+fn poll_answers_deserialization_not_enough() {
+    let json_data = json!([]);
+
+    assert_matches!(
+        from_json_value::<PollAnswers>(json_data),
+        Err(err)
+            if err.is_data()
+            && format!("{}", err) == format!("{}", PollAnswersError::NotEnoughValues)
+    );
```

Comment:
Maybe have an `unwrap_err` here plus separate asserts for `is_data` and `err.to_string() == PollAnswersError::NotEnoughValues.to_string()`? (so the assertion error is more specific if it happens)